packages/blocks/src/database-block/utils.ts

```diff
@@ -19,14 +19,14 @@ function isVisible(elem: HTMLElement) {
 
 export function onClickOutside(
   element: HTMLElement,
-  callback: (element: HTMLElement) => void,
+  callback: (element: HTMLElement) => boolean | void,
```

Comment:
`boolean | void` looks a bit weird, hope for improvement :)

Reply:
yeah，but currently, the onClickOutside function is being used in other places where it doesn't return a value, which may cause conflicts. I haven't thought of a particularly good solution yet👀

Reply:
@luke358 
I refactored `onClickOutside` and added a `reusable` parameter instead of the return `boolean` value of callback.

Reply:
@zqran Yes, using parameters for control is also feasible, good idea!

Reply:
Can you add some tests for this pr?

Reply:
I've added it, you can check it out: #2089